Zend/zend_interfaces_arginfo.h

```diff
@@ -119,3 +119,88 @@ static const zend_function_entry class_InternalIterator_methods[] = {
 	ZEND_ME(InternalIterator, rewind, arginfo_class_InternalIterator_rewind, ZEND_ACC_PUBLIC)
 	ZEND_FE_END
 };
+
+zend_class_entry *register_class_Traversable()
```

Comment:
More of a comment on https://github.com/php/php-src/pull/6289 (didn't see that in the discussions) - If these are included in a header file, I think it would make more sense for the generated code to be `static zend_class_entry *...` instead of ordinary files (e.g. to avoid adding unused copies of the functions in object files or the final program)

It also helps if multiple c files include the same _arginfo.h header?

```c
#define ZEND_BEGIN_ARG_INFO_EX(name, _unused, return_reference, required_num_args)	\
	static const zend_internal_arg_info name[] = { \
		{ (const char*)(zend_uintptr_t)(required_num_args), ZEND_TYPE_INIT_NONE(_ZEND_ARG_INFO_FLAGS(return_reference, 0)), NULL },
```

Reply:
Thanks, Tyson, for the comments! I was also thinking about using `static` declarations, but as far as I remember, Nikita said there is not much difference between non-static and static declarations (it was another PR discussion, so it might make sense in this case. At least including the same arginfo in multiple files can be problematic for sure: https://github.com/php/php-src/pull/6670#discussion_r571420968).

So I'm up to modify gen_stubs.php according to whatever Nikita and you decide. Unfortunately, commenting on this question is above my expertise. P.S. I'm also planning to create a followup PR in order to add `void` to the empty parameter list of the register functions.

Reply:
Using static for this should be fine.

Reply:
Thanks, I'll create a followup PR for switching to `static` class entry generator functions.